### PR TITLE
Fix hostmetricsreceiver/processscraper tests to work for darwin

### PR DESCRIPTION
**Pull request by 8naama**
**Description:** <Describe what has changed.>
- Excluded darwin unsupported metrics from testCases, 'metricsLen', default config
- Excluded from testCases the 'Parent PID Error' test since it's [excluded in the code](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go#L134)
- Added darwin specific test case for Cmdline()

Not sure I took the best approach here (as I also added a function to a DO NOT EDIT file - `generated_metrics.go`), will love your feedback!

**Link to tracking Issue:** [Issue #19141](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19141)

**Testing:** 
Tested locally on Mac to make sure metrics are collected and that running `go test` at the `./hostmetricsreceiver/internal/scraper/processscraper` folder for returns `PASS`

**Documentation:**
added notes for excluding unsupported metrics

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/internal/metadata/generated_metrics.go [hunk 1]
```diff
@@ -48,7 +49,53 @@ type MetricsSettings struct {
 	ProcessThreads             MetricSettings `mapstructure:"process.threads"`
 }
 
+func DeafultDarwinMetricsSettings() MetricsSettings {
```
Comment (mx-psi) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/internal/metadata/generated_metrics.go:52]: This file is autogenerated, we should not modify it. cc @dmitryax, what would be the proper way to handle this?

Comment (8naama) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/internal/metadata/generated_metrics.go]: tested again to remove changes done in that file and should be good now

**Comment by astencel-sumo** [comment]
@8naama, the tests in `process_scraper_test.go` are still being skipped on MacOS, as described in the [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19141):

> the `skipTestOnUnsupportedOS` helper disables tests for metrics that are now supported on Darwin.

These tests need to be run on MacOS as well as on Linux and Windows.

**Comment by astencel-sumo** [comment]
Thanks @8naama, this fixes the tests (just fix the "deafult" typos that the bot has diligently pointed out ;)).

I'm worried that this change introduces even more complexity into this already too complicated test code. Ideally, these tests should be refactored for better maintainability, to have less "if" conditions all over the place. I think this is a task for another PR though.

What do you think @dmitryax?

**Comment by dmitryax** [comment]
I agree. It would be great to simplify the test code overall, but not required for this PR

**Review by dmitryax — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Pull request merged**